gradio/flagging.py

```diff
@@ -206,15 +206,18 @@ def flag(
                         component.label or f"component {idx}"
                     ),
                 )
-                csv_data.append(
-                    component.deserialize(
-                        sample,
-                        save_dir=save_dir,
-                        encryption_key=self.encryption_key,
+                if utils.is_update(sample):
```

Comment:
I think it's strange we treat updates differently from returning a value from a function. At the end of the day, everything is a UI update and I think our api would be easier to maintain if we embraced that. This is similar to #1798. Definitely out of scope for this PR but something I'd like to add to the list of "nice to haves" for  4.0 😂